helix-term/src/commands/typed.rs

```diff
@@ -1225,6 +1225,36 @@ fn lsp_restart(
     Ok(())
 }
 
+fn lsp_stop(
+    cx: &mut compositor::Context,
+    _args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    let (_view, doc) = current!(cx.editor);
+
+    let ls_id = doc
+        .language_server().map(|ls| ls.id())
+        .context("LSP not running for the current document")?;
+
+    let config = doc
+        .language_config()
+        .context("LSP not defined for the current document")?;
+    cx.editor.language_servers.stop(&config);
+
+    cx.editor.documents_mut()
+        .filter(|doc| doc.language_server().map_or(false, |ls| ls.id() == ls_id))
+        .for_each(|doc| {
+            doc.set_language_server(None);
+            log::warn!("XXX STOPPING for {:?}", doc.path());
```

Comment:
You might have forgotten to remove this line before requesting review?

Reply:
https://github.com/helix-editor/helix/pull/5964/commits/b2735d1b6e7188871233c9676692f11bbcf43a20